common/config/environment.go

```diff
@@ -26,6 +26,8 @@ const (
 	DDInfraEnvironment       = "env"
 	DDInfraKubernetesVersion = "kubernetesVersion"
 	DDInfraOSFamily          = "osFamily"
+	DDInfraOSArchitecture    = "osArchitecture"
```

Comment:
This information somewhat conflicts with `instanceType`, we could see if we improve that.
`osAmiId` is also AWS specific and this a generic package, so I'm not sure if it's the right place for it.

Reply:
The concept of the image ID probably exist with other provider. I can maybe rename as `imageID` as a starting point?